### DISCUSSION TITLE
why useLazyAsyncData will trigger twice in `computed`?

### DISCUSSION THREAD

**Discussion by baixiaoyu2997** [nuxt/nuxt#16489] [Questions]
version: `nuxt.rc12`
eg:
```js
const unclaimAllDOBLazyData = computed(() => {
    console.log('=======in computed')
    return useLazyAsyncData(
      'getUnClaimDOB',
      () => {
        console.log('============in fetch')
        return getUnClaimDOB().then((res) => res.unClaim)
      },
      {
        server: false,
        default: () => '0',
        initialCache: false,
      }
    )
  })
```
when i  log `unclaimAllDOBLazyData` in chrome, quick log `unclaimAllDOBLazyData.value`, will show twice  `in computed` and `in fetch`

for now, i switch to use  `computedWithControl` in `vueuse` , change code like this is ok:
```js
const unclaimAllDOBLazyData = computedWithControl(
()=>false,
() => {
    console.log('=======in computed')
    return useLazyAsyncData(
      'getUnClaimDOB',
      () => {
        console.log('============in fetch')
        return getUnClaimDOB().then((res) => res.unClaim)
      },
      {
        server: false,
        default: () => '0',
        initialCache: false,
      }
    )
  })
```

**Comment by danielroe** [comment]
You should never call a composable within a computed. They are meant to be called within a `setup()` function only, or a Nuxt plugin or middleware.

**Reply by vnmtvlv** [reply]
Hey @danielroe why can be useAsyncData() called twice (up to 4-5 times) from just setup?
I created a discussion here https://github.com/nuxt/framework/discussions/8593
![image](https://user-images.githubusercontent.com/97898172/199513641-f5ebc99f-5e5c-4ca4-b44e-64e2155d1329.png)
